src/nrc/utils/schema.py

```diff
@@ -0,0 +1,187 @@
+        """append error serializers"""
+        response_serializers = super().get_response_serializers()
+
+        if self.method == "DELETE":
+            status_code = 204
+            serializer = None
+        elif self._is_create_operation():
+            status_code = 201
+            serializer = response_serializers
+        else:
+            status_code = 200
+            serializer = response_serializers
+
+        responses = {
+            status_code: serializer,
+            **self.get_error_responses(),
+        }
+        return responses
+
+    def _get_response_for_code(
+        self, serializer, status_code, media_types=None, direction="response"
+    ):
+        """
+        choose media types and set descriptions
+        """
+        if not media_types:
+            if int(status_code) >= 400:
+                media_types = [ERROR_CONTENT_TYPE]
+            else:
+                media_types = ["application/json"]
+
+        response = super()._get_response_for_code(
+            serializer, status_code, media_types, direction
+        )
+
+        # add description based on the status code
+        if not response.get("description"):
+            response["description"] = HTTP_STATUS_CODE_TITLES.get(int(status_code), "")
+        return response
+
+    def get_override_parameters(self):
+        """Add request and response headers"""
+        version_headers = self.get_version_headers()
+        content_type_headers = self.get_content_type_headers()
+        location_headers = self.get_location_headers()
+        return version_headers + content_type_headers + location_headers
+
+    def get_version_headers(self) -> list[OpenApiParameter]:
+        return [
+            OpenApiParameter(
+                name=VERSION_HEADER,
+                type=str,
+                location=OpenApiParameter.HEADER,
+                description=_(
+                    "Geeft een specifieke API-versie aan in de context van "
+                    "een specifieke aanroep. Voorbeeld: 1.2.1."
+                ),
+                response=True,
+            )
+        ]
+
+    def get_content_type_headers(self) -> list[OpenApiParameter]:
+        if self.method not in ["POST", "PUT", "PATCH"]:
+            return []
+
+        return [
+            OpenApiParameter(
+                name="Content-Type",
+                type=str,
+                location=OpenApiParameter.HEADER,
+                description=_("Content type of the request body."),
+                enum=["application/json"],
+                required=True,
+            )
+        ]
+
+    def get_location_headers(self) -> list[OpenApiParameter]:
+        return [
+            OpenApiParameter(
+                name="Location",
+                type=OpenApiTypes.URI,
+                location=OpenApiParameter.HEADER,
+                description=_("URL waar de resource leeft."),
+                response=[201],
+            ),
+        ]
```

Comment:
This header is now added to the `Publiceer een Notificatie` operation as well, but I don't think it's actually part of those response headers (not sure why it isn't though):

```
(Pdb) response.headers
{'Content-Type': 'application/json', 'Vary': 'Accept, origin', 'Allow': 'POST, OPTIONS', 'API-version': '1.0.0', 'X-Frame-Options': 'DENY', 'Content-Length': '341', 'X-Content-Type-Options': 'nosniff', 'Referrer-Policy': 'same-origin', 'Cross-Origin-Opener-Policy': 'same-origin'}
```

It is part of the `kanaal_create` operation:

```
(Pdb) response.headers
{'Content-Type': 'application/json', 'Location': 'http://testserver/api/v1/kanaal/84b63731-69b0-4459-8d1d-b240ab1bbd29', 'Vary': 'Accept, origin', 'Allow': 'GET, POST, HEAD, OPTIONS', 'API-version': '1.0.0', 'X-Frame-Options': 'DENY', 'Content-Length': '151', 'X-Content-Type-Options': 'nosniff', 'Referrer-Policy': 'same-origin', 'Cross-Origin-Opener-Policy': 'same-origin'}
```

Reply:
Sorry, on second thought I think it actually makes sense for the `Publiceer een Notificatie` endpoint to not return `Location` headers, because we cannot retrieve indivual notifications at a specific URL (right?). The `data` returned by this operation does not contain a `url`, so the `CreateModelMixin` doesn't work either:

```python
try:
    return {'Location': str(data[api_settings.URL_FIELD_NAME])}
except (TypeError, KeyError):
    return {}
```

Would it be possible to exclude the `get_location_headers` from the `Publiceer een Notificatie` endpoint? :grimacing: